### PR TITLE
netserver/tcp: Add getsockname and getpeername

### PR DESCRIPTION
**Pull request by Dennisbonke**
`OpenSSH` makes use of `sockname()` and `peername()` on tcp sockets. Implement this functionality so `OpenSSH` at least somewhat knows who it is talking to.

### CODE REVIEW

File: servers/netserver/src/ip/tcp4.cpp [hunk 1]
```diff
@@ -410,17 +410,54 @@ struct Tcp4Socket {
...
+		memset(&sa, 0, sizeof(struct sockaddr_in));
+		memset(&addr, 0, sizeof(struct in_addr));
+		sa.sin_family = AF_INET;
+		addr.s_addr = htonl(self->remoteEp_.ipAddress);
+		sa.sin_addr = addr;
+		sa.sin_port = self->remoteEp_.port;
+
+		auto destSize = std::min(sizeof(sockaddr_in), maxAddrLength);
+		memcpy(addrPtr, &sa, destSize);
+		co_return 128;
```
Comment (ArsenArsen) [inline servers/netserver/src/ip/tcp4.cpp:445]: Incorrect return value?

File: servers/netserver/src/ip/tcp4.cpp [hunk 2]
```diff
@@ -410,17 +410,54 @@ struct Tcp4Socket {
...
+		memset(&sa, 0, sizeof(struct sockaddr_in));
+		memset(&addr, 0, sizeof(struct in_addr));
+		sa.sin_family = AF_INET;
+		addr.s_addr = htonl(self->localEp_.ipAddress);
+		sa.sin_addr = addr;
+		sa.sin_port = self->localEp_.port;
+
+		auto destSize = std::min(sizeof(sockaddr_in), maxAddrLength);
+		memcpy(addrPtr, &sa, destSize);
+		co_return 128;
```
Comment (ArsenArsen) [inline servers/netserver/src/ip/tcp4.cpp:427]: Incorrect return value?

These functions should return zero even if the address is trimmed. I could be forgetting how we utilize these return values, though,

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

**Comment by Dennisbonke** [comment]
I believe I've addressed the comments made, ready for another round.

**Review by no92 — APPROVED** [review]

File: servers/netserver/src/ip/tcp4.cpp [hunk 3]
```diff
@@ -410,17 +410,54 @@ struct Tcp4Socket {
...
+		memset(&sa, 0, sizeof(struct sockaddr_in));
+		memset(&addr, 0, sizeof(struct in_addr));
+		sa.sin_family = AF_INET;
+		addr.s_addr = htonl(self->localEp_.ipAddress);
+		sa.sin_addr = addr;
+		sa.sin_port = self->localEp_.port;
+
+		auto destSize = std::min(sizeof(sockaddr_in), maxAddrLength);
+		memcpy(addrPtr, &sa, destSize);
+		co_return destSize;
```
Comment (ArsenArsen) [inline servers/netserver/src/ip/tcp4.cpp:427]:
```
DESCRIPTION
       getsockname() returns the current address to which the socket sockfd is bound, in the buffer pointed to by
       addr.  The addrlen argument should be initialized to indicate the amount of space (in bytes) pointed to by
       addr.  On return it contains the actual size of the socket address.

       The returned address is truncated if the buffer provided is too small; in this case, addrlen will return a
       value greater than was supplied to the call.

RETURN VALUE
       On success, zero is returned.  On error, -1 is returned, and errno is set to indicate the error.
```

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

File: servers/netserver/src/ip/tcp4.cpp [hunk 4]
```diff
@@ -410,17 +410,54 @@ struct Tcp4Socket {
...
+		memset(&sa, 0, sizeof(struct sockaddr_in));
+		memset(&addr, 0, sizeof(struct in_addr));
+		sa.sin_family = AF_INET;
+		addr.s_addr = htonl(self->remoteEp_.ipAddress);
+		sa.sin_addr = addr;
+		sa.sin_port = self->remoteEp_.port;
+
+		auto destSize = std::min(sizeof(sockaddr_in), maxAddrLength);
+		memcpy(addrPtr, &sa, destSize);
+		co_return destSize;
```
Comment (ArsenArsen) [inline servers/netserver/src/ip/tcp4.cpp:445]: ditto

**Review by ArsenArsen — CHANGES_REQUESTED** [review]

**Comment by Dennisbonke** [comment]
Superseeded by #595

**Pull request closed without merging**